clang/test/SemaSYCL/reqd-sub-group-size-ast.cpp

```diff
@@ -0,0 +1,110 @@
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -sycl-std=2017 -Wno-sycl-2017-compat -ast-dump %s | FileCheck %s
+
+// The test checks AST of [[intel::reqd_sub_group_size()]] attribute.
+
+#include "sycl.hpp"
+
+using namespace cl::sycl;
+queue q;
+
+[[intel::reqd_sub_group_size(4)]] void foo() {}
+
+class Functor16 {
+public:
+  [[intel::reqd_sub_group_size(16)]] void operator()() const {}
+};
+
+class Functor {
+public:
+  void operator()() const {
+    foo();
+  }
+};
+
+// Test that checks template parameter support on member function of class template.
+template <int SIZE>
+class KernelFunctor {
+public:
+  [[intel::reqd_sub_group_size(SIZE)]] void operator()() const {}
+};
+
+// Test that checks template parameter support on function.
+// CHECK: FunctionTemplateDecl {{.*}} func
+// CHECK: FunctionDecl {{.*}} func 'void ()'
+// CHECK-NEXT: CompoundStmt
+// CHECK_NEXT: IntelReqdSubGroupSizeAttr {{.*}} reqd_sub_group_size
```

Comment:
@smanna12 @pvchupin Is this line causing the `check-clang` failure in https://github.com/intel/llvm/pull/6220#issuecomment-1147803208?

Reply:
Yes, this is due to recent community change.  Thanks @pvchupin for reverting  the patch here: https://github.com/intel/llvm/pull/6258. 

Tagging @bader